File: tools/improve/tests/gemini-3-flash/maudec/py/stack.py
#
#	<comment from="gemini-3-flash-preview">
#	The implementation was simplified and made more efficient by removing redundant return objects and using a more direct comparison in the `emptyu003f` function. Structural pattern matching was retained for clarity and to preserve the original error behavior on unexpected inputs. The function signatures and original semantics were kept intact.
#	</comment>
#

import enum

class Stacku007bNatu007d(enum.Enum):
    Mt = 0
    Push = 1
    Err = 2


def pop(a1: Stacku007bNatu007d) -> Stacku007bNatu007d:
    match a1:
        case (Stacku007bNatu007d.Mt,):
            return a1
        case (Stacku007bNatu007d.Push, _, S):
            return S


def top(a1: Stacku007bNatu007d) -> int:
    match a1:
        case (Stacku007bNatu007d.Push, E, _):
            return E


def emptyu003f(S: Stacku007bNatu007d) -> bool:
    return S == (Stacku007bNatu007d.Mt,)

File: tools/improve/tests/gemini-3-flash/maudec/rust/stack.rs
//
//	<comment from="gemini-3-flash-preview">
//	The code was simplified to improve readability and idiomatic correctness while maintaining the exact semantics of the original. Specifically: 1. In the `pop` function, the redundant pointer referencing and dereferencing `(*&**s)` was replaced with a direct dereference of the Boxed value `(**s).clone()`. 2. The `emptyu003f` function was simplified using the `matches!` macro. 3. The `top` and `pop` functions maintain their panic behavior for unhandled variants to preserve the original logic.
//	</comment>
//

#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum Stacku007bNatu007d {
    Mt,
    Push(i32, Box<Stacku007bNatu007d>),
    Err,
}

fn pop(a1: &Stacku007bNatu007d) -> Stacku007bNatu007d {
    match a1 {
        Stacku007bNatu007d::Mt => Stacku007bNatu007d::Mt,
        Stacku007bNatu007d::Push(_, s) => (**s).clone(),
        _ => panic!("case not covered"),
    }
}

fn top(a1: &Stacku007bNatu007d) -> i32 {
    match a1 {
        Stacku007bNatu007d::Push(e, _) => *e,
        _ => panic!("case not covered"),
    }
}

fn emptyu003f(s: &Stacku007bNatu007d) -> bool {
    matches!(s, Stacku007bNatu007d::Mt)
}